### PR TITLE
First base R plotting function for Spectra

### PR DESCRIPTION
**Pull request by jorainer**
First proposal of a plotting function for `Spectra` (issue #80):
- Plots each spectrum next to each other.
- Allows to annotate each peak with parameter `labels`.

Use case:
```r
library(Spectra)
mzs <- list(c(13.453421, 43.433122, 46.6653553, 129.111212, 322.24432),
    c(13.452, 43.5122, 129.112, 322.245))
ints <- list(c(4.3412, 12, 8, 34, 23.4),
    c(8, 25, 16, 32))
df <- DataFrame(msLevel = c(1L, 1L), rtime = c(123.12, 124))
df$mz <- mzs
df$intensity <- ints
sp <- Spectra(df)

## Plot each spectrum in a different color
plotSpectra(sp, col = c("green", "blue"))
```

plotSpectra-color

```r
## Use the m/z of each peak as its label
plotSpectra(sp, labels = function(z) format(unlist(mz(z)), digits = 4),
    labelPos = 2, labelOffset = 0.1, labelSrt = -30)
```

plotSpectra-labels

```r
## Use a custom peak annotation/label and color each peak in its own color
sp$label <- list(c("", "A", "B", "C", "D"),
    c("Frodo", "Bilbo", "Peregrin", "Samwise"))
plotSpectra(sp, labels = function(z) unlist(z$label), col = list(1:5, 1:4))
```

plotSpectra-custom-labels

### CODE REVIEW

File: R/plotting-functions.R [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+#'     labelPos = 2, labelOffset = 0.1, labelSrt = -30)
+#' grid()
+#' .plot_single_spectrum(sp, col = "red", type = "p", add = TRUE)
+#'
+#' .plot_single_spectrum(sp, labels = function(z) format(mz(z)[[1]], digits = 5),
+#'     labelPos = 2, labelOffset = 0.1, labelSrt = -30)
+#' grid()
+#'
+#' @noRd
+.plot_single_spectrum <- function(x, xlab = "m/z", ylab = "intensity",
```
Comment (lgatto) [inline R/plotting-functions.R:177]: Is this function the same as `MSnbase:::.plotSingleSpectrum()`? Is so, may be mention is in the source code and/or man page.

File: R/plotting-functions.R [hunk 2]
```diff
@@ -0,0 +1,223 @@
+#' @title Plotting Spectra
```
Comment (lgatto) [inline R/plotting-functions.R:1]: Add a reference to this man page in the `Spectra` documentation.

File: R/plotting-functions.R (hunk 1 above)
Comment (jorainer) [reply R/plotting-functions.R:177]: Actually no - it is a more simple (I guess) function to simply draw a spectrum and have peaks eventually annotated. I was not even aware of the `MSnbase:::.plotSingleSpectrum` function.

File: R/plotting-functions.R (hunk 2 above)
Comment (jorainer) [reply R/plotting-functions.R:1]: No problem adding a reference or adding you also as an author of the function - but to which function should I refer? The `MSnbase::plot`?

File: R/plotting-functions.R (hunk 1 above)
Comment (lgatto) [reply R/plotting-functions.R:177]: OK.

File: R/plotting-functions.R (hunk 2 above)
Comment (lgatto) [reply R/plotting-functions.R:1]: I simply meant a sentence in the Spectra man page that says something along the lines of: _For details on plotting Spectra, see.... _

Comment (jorainer) [reply R/plotting-functions.R:1]: Done (see commit below).

Comment (lgatto) [reply R/plotting-functions.R:1]: Sorry, I wasn't being clear. Will do it myself.

**Pull request merged**